callbacks.h

```diff
@@ -35,17 +36,16 @@
 #include "api/studio/inc/fmod_studio.hpp"
 
 namespace Callbacks {
-
-Mutex *mut;
-
-struct CallbackInfo {
-	Dictionary markerCallbackInfo;
-	Dictionary beatCallbackInfo;
-	Dictionary soundCallbackInfo;
-};
-
-Map<uint64_t, CallbackInfo> eventCallbacks;
-
-FMOD_RESULT F_CALLBACK eventCallback(FMOD_STUDIO_EVENT_CALLBACK_TYPE type, FMOD_STUDIO_EVENTINSTANCE *event, void *parameters);
-
-} // namespace Callbacks
+    struct CallbackInfo {
+		bool markerSignalEmitted = true;
+		bool beatSignalEmitted = true;
+		bool soundSignalEmitted = true;
+		Dictionary markerCallbackInfo;
+		Dictionary beatCallbackInfo;
+		Dictionary soundCallbackInfo;
+	};
+	
+    extern Mutex *mut;
```

Comment:
The mutex doesn't have to be a global variable.

Reply:
Where should we put it ? As a member of the class ? I don't think the CallBack function can access it from there.